app/frontend/src/features/profile/view/ProfilePage.tsx

```diff
@@ -57,39 +57,39 @@ export default function ProfilePage() {
     username ?? (currentUser.data?.username || "")
   );
 
+  const [isRequesting, setIsRequesting] = useState(false);
+
   return (
     <>
       {error && <Alert severity="error">{error}</Alert>}
       {loading ? (
         <CircularProgress />
       ) : user ? (
-        <ProfileUserProvider user={user}>
-          <div className={classes.root}>
-            <Overview user={user} />
-            <Card className={classes.detailsCard}>
-              <TabContext value={currentTab}>
-                <TabBar
-                  value={currentTab}
-                  setValue={setCurrentTab}
-                  labels={SECTION_LABELS}
-                  ariaLabel="tabs for user's details"
-                />
-                <TabPanel classes={{ root: classes.tabPanel }} value="about">
-                  <About user={user} />
-                </TabPanel>
-                <TabPanel value="home">
-                  <Home user={user}></Home>
-                </TabPanel>
-                <TabPanel
-                  classes={{ root: classes.tabPanel }}
-                  value="references"
-                >
-                  <References user={user} />
-                </TabPanel>
-              </TabContext>
-            </Card>
-          </div>
-        </ProfileUserProvider>
+        <div className={classes.root}>
+          <Overview user={user} setIsRequesting={setIsRequesting} />
+          <Card className={classes.detailsCard}>
+            <TabContext value={currentTab}>
+              <TabBar
+                value={currentTab}
+                setValue={setCurrentTab}
+                labels={SECTION_LABELS}
+                ariaLabel="tabs for user's details"
+              />
+              {isRequesting && (
+                <NewHostRequest user={user} setIsRequesting={setIsRequesting} />
```

Comment:
Using [`Collapse`](https://material-ui.com/components/transitions/#collapse) here would be nice